custom/icds_reports/utils/aggregation_helpers/distributed/agg_dashboard_acitivity.py

```diff
@@ -44,29 +44,69 @@ def get_dashboard_users(self):
+        """
+        :return: Returns a dict containing location_id as key and its info(loc_level,parents) as value
+                eg: {block_loc_id1: {'loc_level':3, parents:{
+                                                    'district_id': district_loc_id1,
+                                                    'state_id': state_loc_id1
+                                                }
+                                    }
+                    }
+        """
+        from custom.icds_reports.models.aggregate import AwcLocation
+        locations = AwcLocation.objects.filter(aggregation_level=3).values('state_id','district_id','block_id')
+
+        transformed_locations = dict()
+
+        for loc in locations:
+            state_id = loc['state_id']
+            district_id = loc['district_id']
+            block_id = loc['block_id']
+
+            if state_id not in transformed_locations:
+                transformed_locations[state_id] = {'loc_level': 1}
+
+            if district_id not in transformed_locations:
+                transformed_locations[district_id] = {'loc_level': 2,
+                                                      'parents': {
+                                                          'state_id': state_id
+                                                      }
+                                                      }
+            if block_id not in transformed_locations:
+                transformed_locations[block_id] = {'loc_level': 3,
+                                                   'parents': {
+                                                       'district_id': district_id,
+                                                       'state_id': state_id
+                                                      }
+                                                   }
+        return transformed_locations
+
     def get_user_locations(self):
         user_locations = list()
+        print(self.get_dashboard_users)
         for username in self.get_dashboard_users:
             user = CommCareUser.get_by_username(username)
-            loc = user.sql_location
 
             state_id, district_id, block_id, user_level = None, None, None, None
 
-            if loc and loc.location_type.name == 'state':
-                state_id = loc.location_id
-                district_id = 'All'
-                block_id = 'All'
-                user_level = 1
-            elif loc and loc.location_type.name == 'district':
-                district_id = loc.location_id
-                state_id = loc.get_ancestor_of_type('state').location_id
-                block_id = 'All'
-                user_level = 2
-            elif loc and loc.location_type.name == 'block':
-                block_id = loc.location_id
-                district_id = loc.get_ancestor_of_type('district').location_id
-                state_id = loc.get_ancestor_of_type('state').location_id
-                user_level = 3
+            if user.location_id:
```

Comment:
what is the difference between `location_id` and `sql_location` and what happens if the user is assigned to multiple locations?

Reply:
Hey, sql_location is a `@property` of the class which returns the object of SQLLocation which intern hitting the DB. and `location_id` is just the string location_id assigned to the user. which is the primary location Id. 
`What happens if the user is assigned to multiple locations?` This will use the primary location out of it. 
Does it actually happen on ICDS? I know we don't technically restrict from doing it on UI. But do we get any real user with multiple locations assigned on ICDS?

Reply:
Confirmed on Skype. Only valid user with multiple locations will be CPMU user which is not a part of this report. So in this report users will have only one location assigned